store/rootmulti/store.go

```diff
@@ -786,27 +787,27 @@ func flushMetadata(db dbm.DB, version int64, cInfo commitInfo, pruneHeights []in
 // SimpleHashFromMap computes a merkle tree from sorted map and returns the merkle
 // root.
 func SimpleHashFromMap(m map[string][]byte) []byte {
-	mm := newMerkleMap()
+	mm := sdkmaps.NewMerkleMap()
 	for k, v := range m {
-		mm.set(k, v)
+		mm.Set(k, v)
 	}
 
-	return mm.hash()
+	return mm.Hash()
 }
 
 // SimpleProofsFromMap generates proofs from a map. The keys/values of the map will be used as the keys/values
 // in the underlying key-value pairs.
 // The keys are sorted before the proofs are computed.
 func SimpleProofsFromMap(m map[string][]byte) (rootHash []byte, proofs map[string]*merkle.SimpleProof, keys []string) {
-	sm := newSimpleMap()
+	sm := sdkmaps.NewSimpleMap()
 	for k, v := range m {
 		sm.Set(k, v)
 	}
 	sm.Sort()
-	kvs := sm.kvs
+	kvs := sm.Kvs
 	kvsBytes := make([][]byte, len(kvs))
 	for i, kvp := range kvs {
-		kvsBytes[i] = KVPair(kvp).Bytes()
+		kvsBytes[i] = sdkmaps.KVPair(kvp).Bytes()
 	}
 
 	rootHash, proofList := merkle.SimpleProofsFromByteSlices(kvsBytes)
```

Comment:
Okay, this is actually getting tangled up.

We have a sdk SimpleMap, which does pre-processing and feed into tendermint SimpleProofFromByteSlices (which returns a ton of proofs, many not needed). Then we take those, select the 1 or 2 we want and convert them into ics23 format.

:exploding_head: 

Why not just have sdk.SimpleMap return the ics23 proofs directly?

If you are so adverse to adding this logic into tendermint, just pull it all into the sdk. Then we can replace `merkle.SimpleProofsFromByteSlices` with `SimpleMap.GenertaeProof(key)` which returns and `ics23.ExistenceProof` directly.

Reply:
It may be beneficial to do this in a separate PR. I will create a issue for it to not lose track of the conversation. Does that work?

Reply:
Yea, definitely convoluted right now.

Making the ics proofs return directly is definitely next step for both simple proof here and in the iavl.

@marbar3778 could you make an issue here? Ill make a corresponding issue on the iAVL